src/Sylius/Bundle/ApiBundle/spec/DataProvider/OrderPaymentMethodsSubresourceDataProviderSpec.php

```diff
@@ -0,0 +1,184 @@
+        PaymentRepositoryInterface $paymentRepository,
+        OrderInterface $order,
+        PaymentInterface $payment
+    ): void {
+        $context['subresource_identifiers'] = ['id' => '69', 'payments' => '420'];
+
+        $orderRepository->findCartByTokenValue($context['subresource_identifiers']['id'])->willReturn($order);
+        $paymentRepository->find($context['subresource_identifiers']['payments'])->willReturn($payment);
+
+        $order->hasPayment($payment)->willReturn(false);
+
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('getSubresource', [
+                PaymentMethodInterface::class,
+                [],
+                $context,
+                Request::METHOD_GET,
+            ])
+        ;
+    }
+
+    function it_throws_an_exception_if_payment_does_not_match_order(
+        OrderRepositoryInterface $orderRepository,
+        PaymentRepositoryInterface $paymentRepository,
+        OrderInterface $order,
+        PaymentInterface $payment
+    ): void {
+        $context['subresource_identifiers'] = ['id' => '69', 'payments' => '420'];
+
+        $orderRepository->findCartByTokenValue($context['subresource_identifiers']['id'])->willReturn($order);
+        $paymentRepository->find($context['subresource_identifiers']['payments'])->willReturn($payment);
+
+        $order->hasPayment($payment)->willReturn(false);
+
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('getSubresource', [
+                PaymentMethodInterface::class,
+                [],
+                $context,
+                Request::METHOD_GET,
+            ])
+        ;
+    }
+
+    function it_returns_empty_array_if_cart_does_not_have_payments(
+        OrderRepositoryInterface $orderRepository,
+        PaymentRepositoryInterface $paymentRepository,
+        OrderInterface $order,
+        PaymentInterface $payment
+    ): void {
+        $context['subresource_identifiers'] = ['id' => '69', 'payments' => '420'];
+
+        $orderRepository->findCartByTokenValue($context['subresource_identifiers']['id'])->willReturn($order);
+        $paymentRepository->find($context['subresource_identifiers']['payments'])->willReturn($payment);
+
+        $order->hasPayment($payment)->willReturn(true);
+        $order->hasPayments()->willReturn(false);
+
+        $this
+            ->getSubresource(
+                PaymentMethodInterface::class,
+                [],
+                $context,
+                Request::METHOD_GET
+            )
+            ->shouldReturn([])
+        ;
+    }
+
+    function it_returns_order_payment_methods(
+        OrderRepositoryInterface $orderRepository,
+        PaymentRepositoryInterface $paymentRepository,
+        OrderInterface $order,
+        PaymentInterface $payment,
+        PaymentMethodsResolverInterface $paymentMethodsResolver,
+        PaymentMethodInterface $paymentMethod
```

Comment:
```suggestion
        OrderRepositoryInterface $orderRepository,
        PaymentRepositoryInterface $paymentRepository,
        PaymentMethodsResolverInterface $paymentMethodsResolver,
        OrderInterface $order,
        PaymentInterface $payment,
        PaymentMethodInterface $paymentMethod
```
First arguments should be the same as in `let` method